src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFile_Native.base.cs

```diff
@@ -0,0 +1,261 @@
+﻿#nullable enable
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.Storage;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_Storage
+{
+	public abstract class Given_StorageFile_Native_Base
+	{
+		protected abstract Task<StorageFolder> GetRootFolderAsync();
+
+		protected virtual Task CleanupRootFolderAsync() => Task.CompletedTask;
+
+		[TestMethod]
+		public async Task When_CreateFile_Name_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var fileName = GetRandomTextFileName();
+			StorageFile? createdFile = null;
+			try
+			{
+				createdFile = await rootFolder.CreateFileAsync(fileName);
+				Assert.AreEqual(fileName, createdFile.Name);
+			}
+			finally
+			{
+				if (createdFile != null)
+				{
+					await createdFile.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+#if WINDOWS_UWP
+		[Ignore("On UWP, DisplayName sometimes returns the name with, and sometimes without the extension.")]
```

Comment:
Compare with `StorageFile.Name` instead with Path.GetFileNameWithoutExtension()?

Reply:
Unfortunately it is quite unreliable 🤔 . Sometimes it includes the extension, sometimes not. In Uno, I made it return without extension always, but UWP is not reliable in this regard. People have tried to solve this on SO several times 😀 https://stackoverflow.com/questions/50619860/how-to-make-the-storagefile-displayname-getter-exclude-the-filetype-systematical